### PR TITLE
Fix #4106: Showing color preview in Edit Profile

### LINKED ISSUE #4106: Bad coloring on follow button
**Describe the bug**

When reading an article, sometimes the button is not clear on what it says given the colors.

**To Reproduce**

I found it on this article: https://dev.to/jmfayard/how-kotlin-makes-editing-your-gradle-build-less-frustrating-232l

Just going there, without dark mode or anything, I'm not sure if I'm following or not this particular user.

**Expected behavior**

Text with the follow status for this user should be visible or selectable

**Screenshots**

image

**Desktop (please complete the following information):**

- OS: MacOS
- Browser: Firefox
- Version: 69.0.1

### PR DESCRIPTION
**Pull request by saurabhdaware**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Fixes #4106. 

The current color-picker does not show a preview so it is hard to guess the color contrast. I added a preview (screenshot added below) which shows your selected colors.

Also, Added a `<input type="color">` which makes it is easy to select the color.

Since there are other pages that depend on 'color-picker' class I did not change anything in `colorPicker.js` or any of the styles of current color-picker. I created a new file `colorPreview.js` and some new classes in html. 

### Mobile
Mobile screenshot of color preview feature

### Desktop
Desktop Screenshot of color preview feature

### Recording
Recording of color preview feature

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
My first PR To DEV!
I was not very sure if I could do this or not since DEV's codebase is probably the largest codebase I've ever worked on but now I'm super proud of myself to complete this 🌻

Me showing this code to everyone :
Elsa showing magic to Anna

### CODE REVIEW

File: app/javascript/packs/colorPreview.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// Event Listeners
+colorField.bgColor.addEventListener('input', watchColorFields);
+colorField.textColor.addEventListener('input', watchColorFields);
+textField.bgColor.addEventListener('keyup', watchTextFields);
+textField.textColor.addEventListener('keyup', watchTextFields);
+
+// on init
+updateColorFields();
+updatePreview();
+preview.style.display = 'inline-block';
```
Comment (rhymes) [inline app/javascript/packs/colorPreview.js]: This should probably be inside the `window.InstantClick.on('change', () => {` block, see the colorPicker.js

Comment (saurabhdaware) [reply app/javascript/packs/colorPreview.js]: Hi I added it inside InstantClick change event. Also, just adding it inside InstantClick block didn't work so had to call it from outside as well. Even in `colorPicker.js` there is similar code outside the block and inside the change block. 

***colorPicker.js***
```js
const colorInputs = document.getElementsByClassName('color-picker');
const colorPalettes = document.getElementsByClassName('color-palette');

for (let i = 0; i < colorInputs.length; i += 1) {
  makeColorPickerGo(colorInputs[i], colorPalettes[i]);
}

window.InstantClick.on('change', () => {
  const colorInputElements = document.getElementsByClassName('color-picker');
  const colorPaletteElements = document.getElementsByClassName('color-palette');

  for (let i = 0; i < colorInputElements.length; i += 1) {
    makeColorPickerGo(colorInputElements[i], colorPaletteElements[i]);
  }
});

```

Just wondering when exactly InstantClick is fired? It is used in a lot of files so hard to figure out

**Comment by rhymes** [comment]
@saurabhdaware can you take a look at the code climate issues as well just in case? I think it's complaining that `updateColorFields` and `updateTextFields` are basically doing the same exact thing.

**Comment by saurabhdaware** [comment]
Not sure why it is showing the issue in it. `updateColorField()` Updates the color fields to be in sync with text fields and I call it when text fields are changed. and `updateTextField()` updates text fields to be in sync with color fields and they are called when color input is changed. can't really think of any other way of doing it.

**Comment by saurabhdaware** [comment]
Cool, thank you @rhymes ! and yeah we can pass parameters and get around code climate I guess. This way just makes it easier for me to update the fields whenever necessary. I'll keep it in mind for the future though. thanks :D

**Pull request merged**